src/Views/BucketPanel/EntryDetail.tsx

```diff
@@ -123,7 +139,95 @@ export default function EntryDetail(props: Readonly<Props>) {
     }
   };
 
+  const handleUpload = async (values: any) => {
+    setIsUploadLoading(true);
+    try {
+      if (!uploadFile) {
+        throw new Error("Please select a file to upload");
+      }
+
+      // Check write permissions
+      if (!props.permissions?.write) {
+        throw new Error(
+          "You don't have permission to write to this bucket. Please check your token permissions.",
+        );
+      }
+
+      // First verify we can connect to the bucket
+      try {
+        const bucket = await props.client.getBucket(bucketName);
+
+        // Verify bucket is accessible
+        await bucket.getInfo();
+      } catch (error) {
+        const connError = error as Error;
+        throw new Error(`Cannot connect to bucket: ${connError.message}`);
+      }
+
+      const bucket = await props.client.getBucket(bucketName);
+
+      // Read the file as an ArrayBuffer for binary data
+      const arrayBuffer = await uploadFile.arrayBuffer();
+
+      const writer = await bucket.beginWrite(entryName, {
+        contentType: uploadFile.type || "application/octet-stream",
+        labels: labels.reduce(
+          (acc, label) => ({
+            ...acc,
+            [label.key.trim()]: label.value,
+          }),
+          {},
+        ),
+      });
+
+      try {
+        // Convert Uint8Array to Buffer
+        const buffer = Buffer.from(arrayBuffer);
+
+        // Write the binary data
+        await writer.write(buffer);
+
+        setIsUploadModalVisible(false);
+        uploadForm.resetFields();
+        setUploadFile(null);
+        setLabels([]);
+
+        // Add a small delay before refreshing
+        setTimeout(() => {
+          getRecords(start, end, limit);
+        }, 1000);
+      } catch (error) {
+        const writeError = error as Error;
+        console.log("Write error:", writeError);
+        throw new Error(`Failed to write file: ${writeError.message}`);
+      }
+    } catch (error) {
+      const err = error as Error;
+      console.log("Upload error:", err);
+      setUploadError(
+        err.message ||
+          "Failed to upload file. Please check your connection and try again.",
+      );
+    } finally {
+      setIsUploadLoading(false);
+    }
+  };
+
   useEffect(() => {
+    // Verify connection when component mounts
+    const verifyConnection = async () => {
```

Comment:
I don't see it's necessary. Could you explain your motivation for additional check?